### PR TITLE
Enhanced README with Troubleshooting Section.

### PR DESCRIPTION
**Pull request by vaishnavipal1869**
## Description #80 
- The **addition of the troubleshooting section** aims to enhance the user experience by providing immediate solutions to common problems encountered while using the app. This **proactive approach** empowers users to troubleshoot issues independently, thereby reducing frustration and the need for support inquiries.

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

## Screenshot of successful build.
- Unfortunately, I do not have a screenshot of the successful Gradle build as my contribution was focused on improving documentation.

## Screenshots / Screen Recordings
- Please refer to the updated README for the troubleshooting section.

## Request for Review
- I believe I have completed all the required steps for this pull request. I would greatly appreciate it if you could take a moment to review my changes and provide your feedback. Thank you!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,6 +41,46 @@ Interact with people within a 100-meter radius! Get real-time locations of nearb
 
 Time to mingle, chat, and rate! Have fun exploring the world around you. 🌍🎉
 
+## Troubleshooting 🤔
+
+If you encounter any issues while using the app, this section will help you find quick solutions to common problems.
+
+### 1. App Crashes on Device Configuration Changes
```
Comment (Vishwajith-Shettigar) [inline README.md]: This shouldn't be in Troubleshooting section.

File: README.md [hunk 2]
```diff
@@ -41,6 +41,46 @@ Interact with people within a 100-meter radius! Get real-time locations of nearb
...
+- Ensure that `android:configChanges="orientation|screenSize"` is added in the AndroidManifest.xml for the affected activities.
+- Consider using ViewModel or handling configuration changes appropriately in the activity lifecycle.
+
+### 2. User's Location is Not Detected
+**Issue:** The app is not able to detect the user's location.  
+**Solution:**  
+- Make sure that location permissions are granted in the device settings.
+- If the location is off, prompt the user to enable location services.
+
+### 3. Profile Pictures Still Visible Despite Disabling
```
Comment (Vishwajith-Shettigar) [inline README.md]: This is an issue which will be resolved soon. Putting it in this section doesn't make any sense ig.

File: README.md [hunk 3]
```diff
@@ -41,6 +41,46 @@ Interact with people within a 100-meter radius! Get real-time locations of nearb
...
+**Issue:** Profile pictures are still visible on the map or chat screen, even after setting them to be disabled.  
+**Solution:**  
+- Check the API response to ensure that the `profilePictureVisible` flag is properly updated.
+- Ensure that the visibility logic is correctly handled in the frontend component.
+
+### 4. Red Stroke Not Showing in Default Image on Map
```
Comment (Vishwajith-Shettigar) [inline README.md]: Same here.

File: README.md [hunk 4]
```diff
@@ -41,6 +41,46 @@ Interact with people within a 100-meter radius! Get real-time locations of nearb
...
+**Issue:** Red stroke does not appear in the default image on the map.  
+**Solution:**  
+- Check if the image loading logic is correctly implemented in the Map component.
+- Ensure that the red stroke is added as an overlay when the map is rendered.
+
+### 5. Online Status Not Working Properly
```
Comment (Vishwajith-Shettigar) [inline README.md]: Same here

**Review by Vishwajith-Shettigar — CHANGES_REQUESTED** [review]
In troubleshooting section, only add some common build failures contributors face.

**Comment by Vishwajith-Shettigar** [comment]
@vaishnavipal1869  you can add below to troubleshooting guide.

https://github.com/Vishwajith-Shettigar/NextGen/issues/75#issuecomment-2408345836

https://github.com/Vishwajith-Shettigar/NextGen/issues/75#issuecomment-2406603579 ( Please have a quick chat with @MiRa202321 to understand the problem and solution she came up with.

https://github.com/Vishwajith-Shettigar/NextGen/discussions/47#discussion-7274929

**Comment by vaishnavipal1869** [comment]
I can incorporate the requested changes and make adjustments as needed.
Should I proceed with that?

**Comment by MiRa202321** [comment]
@vaishnavipal1869 Hi there! I had to ask that why is the map blank? Can it be because of some fault while setting up? I have allowed the permissions to access my location. Also, I have added the map's api.
Screenshot 2024-10-12 111458

**Comment by Vishwajith-Shettigar** [comment]
> @vaishnavipal1869 Hi there! I had to ask that why is the map blank? Can it be because of some fault while setting up? I have allowed the permissions to access my location. Also, I have added the map's api. Screenshot 2024-10-12 111458

Reinstall and give all permissions, and turn on device location. It should work

**Comment by vaishnavipal1869** [comment]
image

@Vishwajith-Shettigar,

Hello! I have made the requested changes to the README file. However, I don’t see an option to create a new pull request. Could you please advise on how to proceed?

Thank you!

**Comment by Vishwajith-Shettigar** [comment]
> image
> 
> @Vishwajith-Shettigar,
> 
> Hello! I have made the requested changes to the README file. However, I don’t see an option to create a new pull request. Could you please advise on how to proceed?
> 
> Thank you!

You dont need to create new PR !

**Comment by vaishnavipal1869** [comment]
hey @Vishwajith-Shettigar can you please look into the updated readme and assign me with the points???

**Comment by Vishwajith-Shettigar** [comment]
> @vaishnavipal1869 you can add below to troubleshooting guide.
> 
> [#75 (comment)](https://github.com/Vishwajith-Shettigar/NextGen/issues/75#issuecomment-2408345836)
> 
> [#75 (comment)](https://github.com/Vishwajith-Shettigar/NextGen/issues/75#issuecomment-2406603579) ( Please have a quick chat with @MiRa202321 to understand the problem and solution she came up with.
> 
> [#47 (comment)](https://github.com/Vishwajith-Shettigar/NextGen/discussions/47#discussion-7274929)

Please add the above to the troubleshooting guide and retain the rest of the old troubleshooting guide, except for the parts I mentioned https://github.com/Vishwajith-Shettigar/NextGen/pull/89#discussion_r1797545554 , https://github.com/Vishwajith-Shettigar/NextGen/pull/89#discussion_r1797545800, https://github.com/Vishwajith-Shettigar/NextGen/pull/89#discussion_r1797545820 , https://github.com/Vishwajith-Shettigar/NextGen/pull/89#discussion_r1797545850

**Comment by vaishnavipal1869** [comment]
@MiRa202321 can you please explain the problem and the solution you came up with

**Comment by vaishnavipal1869** [comment]
hello sir @Vishwajith-Shettigar could you please review the requested chnages?, thank you!!

File: README.md [hunk 5]
```diff
@@ -41,6 +41,48 @@ Interact with people within a 100-meter radius! Get real-time locations of nearb
...
+- **Solution:** Review the `build.gradle` files and check for any version conflicts. Updating the library versions to be compatible might solve the issue.
+
+### 4. Insufficient Memory During Build
+- **Issue:** Running out of memory while building the project.
+- **Solution:** Increase the heap size in the `gradle.properties` file. Try allocating more memory to the Gradle build by updating:
+  ```properties
+  org.gradle.jvmargs=-Xmx2048m
+  ```
+
+### 5. Network Issues
```
Comment (Vishwajith-Shettigar) [inline README.md]: Remove this.

**Review by Vishwajith-Shettigar — COMMENTED** [review]
https://github.com/Vishwajith-Shettigar/NextGen/discussions/47  add this.

**Comment by vaishnavipal1869** [comment]
@Vishwajith-Shettigar is this okay now ?

File: README.md [hunk 6]
```diff
@@ -41,6 +41,48 @@ Interact with people within a 100-meter radius! Get real-time locations of nearb
...
+- **Issue:** Running out of memory while building the project.
+- **Solution:** Increase the heap size in the `gradle.properties` file. Try allocating more memory to the Gradle build by updating:
+  ```properties
+  org.gradle.jvmargs=-Xmx2048m
+  ```
+
+### 5. Clarification on SERVER_URL in secrets.properties from #47
+**Issue:** Uncertainty about the value to use for SERVER_URL.
+**Solution:** If you are not working on the video call feature, please add any random characters as a workaround, e.g., SERVER_URL=pbdwkw. Just for your information, SERVER_URL holds the URL of the video call server; you can find the repository for reference.
+You can find the repository for reference [here](https://github.com/Vishwajith-Shettigar/video-call-server-node.js).
```
Comment (Vishwajith-Shettigar) [inline README.md:84]: Repeated sentence.

Comment (Vishwajith-Shettigar) [reply README.md:84]: you can find the repository for reference.
You can find the repository for reference [here]

**Comment by vaishnavipal1869** [comment]
@Vishwajith-Shettigar sir i have successfully removed the duplicate sentence, would you please look into it.

**Comment by Vishwajith-Shettigar** [comment]
Thanks @vaishnavipal1869

**Pull request merged**